packages/gatsby-cli/src/reporter/loggers/ink/components/messages.tsx

```diff
@@ -0,0 +1,70 @@
+import React from "react"
+import { Box, Color, ColorProps } from "ink"
+
+import { ActivityLogLevels, LogLevels } from "../../../constants"
+
+const ColorSwitcher: React.FC<ColorProps> = ({ children, ...props }) => (
+  <Color {...props}>{children}</Color>
+)
+
+const createLabel = (text: string, color: string): React.FC<ColorProps> => (
+  ...props
+): JSX.Element => (
+  <ColorSwitcher {...{ [color]: true, ...props }}>{text}</ColorSwitcher>
+)
+
+type ValueOf<T> = T[keyof T]
```

Comment:
I used Generics to get values of objects as a type alias in order to ensure type safety below.

Reply:
I think if we change the LogLevels to enums this typing will simplify. I'd like to get away from this if we can